Hackathon_Converter/MainForm.cs

```diff
@@ -51,63 +55,495 @@ private void OnAddFilesButtonClick(object sender, EventArgs e)
                     fileDataGrid.Rows.Add(new String[] { fileInfo.FullName });
                 }
             }
+
+            Show_Conversion_Page();
+            this.Btn_Convert.Enabled = true;
+            
         }
 
         private void OnConvertButtonClick(object sender, EventArgs e)
         {
+            //Implement for dropdown options setConfigObject();
+
             Stream htmlStream;
+
+            StringBuilder fileName = new StringBuilder();
+            fileName.Append(FileNameInput.Text.Trim());
+            if (fileName.ToString().Length > 0)
+                fileName.Append(".html");
+            else
+                fileName.Append("out.html");
+            
+
             SaveFileDialog saveFileDialog = new SaveFileDialog
             {
-                FileName = "out.html",
+                FileName = fileName.ToString(),
                 Filter = "HTML files (*.html)|*.html|All files (*.*)|*.*",
                 FilterIndex = 1,
-                RestoreDirectory = true
+                RestoreDirectory = false
             };
 
             if (saveFileDialog.ShowDialog() != DialogResult.OK)
             {
                 return;
+                
             }
             if ((htmlStream = saveFileDialog.OpenFile()) == null)
+            {
+                Show_Error_Page();
+            }
+            else
+            { 
+                btn_AddFiles.Enabled = false;
+                fileDataGrid.Enabled = false;
+                Show_Loading_Page();
+
+
+                // Does not parse through section headers yet
+                var parser = new USFMToolsSharp.USFMParser(new List<string> { "s5","s","s2","s3","s4" });
+
+                formatConfig();
+                //Configure Settings -- Spacing ? 1, Column# ? 1, TextDirection ? L2R 
+                var renderer = new USFMToolsSharp.HtmlRenderer(configHTML);
+
+            
+                // Added ULB License and Page Number
+                renderer.FrontMatterHTML = GetLicenseInfo();
+                renderer.InsertedFooter = GetFooterInfo();
+
+                var usfm = new USFMToolsSharp.Models.Markers.USFMDocument();
+
+                var progress = fileDataGrid.RowCount - 1;
+                var progressStep = 0;
+
+                foreach (DataGridViewRow row in fileDataGrid.Rows)
+                {
+                    var cell = row.Cells[0];
+                    if (cell.Value == null)
+                    {
+                        continue;
+                    }
+                    var filename = cell.Value.ToString();
+                    var text = File.ReadAllText(filename);
+                    usfm.Insert(parser.ParseFromString(text));
+
+                    progressStep++;
+                    LoadingBar.Value = (int)(progressStep / (float)progress * 100);
+                }
+
+                var html = renderer.Render(usfm);
+                var bytes = Encoding.UTF8.GetBytes(html);

```

Comment:
reduced to WriteAllText function

Reply:
Very nice

Reply:
For some reason WriteAllText gives me a System.Io.IOException. 

Says the 'The process cannot access the file because it is being used by another process.'

Reply:
WriteAllText works fine now! Thanks for the help!